### PR TITLE
Export Environment of chakra-ui

### PR DESCRIPTION
**Pull request by benits**
### @chakra-ui/react-env
React component and hook for handling window and document object in iframe or ssr environment

```
This is an internal utility, not intended for public usage.

```
To get it working, you need to wrap your app in EnvironmentProvider and call the useEnvironment hook anywhere in your app to get access to the correct window and document.

## Proposed changes
In this PR EnvironmentProvider,  EnvironmentProviderProps and useEnvironment were added that were exiting in the chakra-ui but were not being exported in the rock-ui.

Components added:
- EnvironmentProvider
- EnvironmentProviderProps
- useEnvironment

## Does this close any currently open issues?
- Export Environment of chakra-ui- https://taiga6.rockcontent.com/project/rs-frontend/task/56

## Pull request checklist
_Put an `x`  in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
Please check if your PR fulfills the following requirements:
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Pull request type

What types of changes does your code introduce to Rock-ui?
_Put an `x`  in the boxes that apply_
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe): 

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
N/A

 Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

File: src/components/env/stories/env.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+const Portal = ({ children }: PortalProps) => {
+  const env = useEnvironment();
+  return createPortal(children, env.document.body);
+};
+
+export const WithIframe = () => {
+  return (
+    <Box className="App">
+      <h1>Hello CodeSandbox</h1>
```
Comment (fernandacanto) [inline src/components/env/stories/env.stories.tsx:28]: Acho que é melhor usar o componente "Heading" aqui, para que o exemplo utilize os componentes do chakra.

File: src/components/env/stories/env.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <h2>Start editing to see some magic happen!</h2>
+      <Portal>Outside iframe</Portal>
+      <Frame style={{ background: 'yellow' }}>
+        <EnvironmentProvider>
+          <span>Welcome home</span>
```
Comment (fernandacanto) [inline src/components/env/stories/env.stories.tsx:33]: Aqui da pra usar um "<Box as="span">.

File: src/components/env/stories/env.stories.tsx (hunk 1 above)
Comment (benits) [reply src/components/env/stories/env.stories.tsx:28]: show demais, estou ajustando :)

File: src/components/env/stories/env.stories.tsx (hunk 2 above)
Comment (benits) [reply src/components/env/stories/env.stories.tsx:33]: tem algum motivo especifico?

Comment (benits) [reply src/components/env/stories/env.stories.tsx:33]: o proprio linter do projeto que formatou para ''.

Comment (benits) [reply src/components/env/stories/env.stories.tsx:33]: Podemos adicionar a regra pra tudo ser double ou single quote

Comment (fernandacanto) [reply src/components/env/stories/env.stories.tsx:33]: É que meu comentário não foi completo... kkkk
Eu sugeri usar um Box com o as span, o quote está ok assim! ;)

Comment (benits) [reply src/components/env/stories/env.stories.tsx:33]: show, vou atualizar aqui

Comment (benits) [reply src/components/env/stories/env.stories.tsx:33]: done

**Pull request merged**